zap/src/main/java/org/zaproxy/zap/view/MultipleRegexesOptionsPanel.java

```diff
@@ -76,6 +83,50 @@ public MultipleRegexesOptionsPanel(Dialog owner, RegexesTableModel model) {
 
         this.owner = owner;
         getTable().setSortOrder(0, SortOrder.ASCENDING);
+        JButton importButton =
+                new JButton(Constant.messages.getString("context.include.import.label"));
+        importButton.addActionListener(e -> importButtonClicked(e));
+        addButton(importButton);
+    }
+
+    public void importButtonClicked(ActionEvent event) {
+        File file = selectImportFile();
+        if (file != null) {
+            List<String> regexes = getRegexes();
+
+            String line = null;
+            try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+                while ((line = br.readLine()) != null) {
+                    String trimmedRegex = line.trim();
+                    if (!regexes.contains(trimmedRegex)) {
+                        Pattern.compile(trimmedRegex, Pattern.CASE_INSENSITIVE);
+                        regexes.add(trimmedRegex);
+                    }
+                }
+            } catch (IOException | PatternSyntaxException e) {
+                View.getSingleton()
+                        .showWarningDialog(
+                                Constant.messages.getString(
+                                        "context.include.import.error", e.getLocalizedMessage()));
+                return;
+            }
+            setRegexes(regexes);
+        }
+    }
+
+    private File selectImportFile() {
+        JFileChooser chooser = new JFileChooser(Constant.getContextsDir());
```

Comment:
Not sure the contexts dir is the most appropriate, that's used for the whole contexts. Don't specify any?

Reply:
Removed path.